addons/mail/static/tests/core/search_highlight.test.js

```diff
@@ -11,94 +12,100 @@ import {
     startServer,
     triggerHotkey,
 } from "@mail/../tests/mail_test_helpers";
+
 import { describe, expect, test } from "@odoo/hoot";
-import { serverState } from "@web/../tests/web_test_helpers";
+import { markup } from "@odoo/owl";
 
-import { HIGHLIGHT_CLASS, searchHighlight } from "@mail/core/common/message_search_hook";
+import { serverState } from "@web/../tests/web_test_helpers";
 
 describe.current.tags("desktop");
 defineMailModels();
 
 test("Search highlight", async () => {
     const testCases = [
         {
-            input: "test odoo",
+            input: markup("test odoo"),
             output: `test <span class="${HIGHLIGHT_CLASS}">odoo</span>`,
             searchTerm: "odoo",
         },
         {
-            input: '<a href="https://www.odoo.com">https://www.odoo.com</a>',
+            input: markup('<a href="https://www.odoo.com">https://www.odoo.com</a>'),
             output: `<a href="https://www.odoo.com">https://www.<span class="${HIGHLIGHT_CLASS}">odoo</span>.com</a>`,
             searchTerm: "odoo",
         },
         {
-            input: '<a href="https://www.odoo.com">Odoo</a>',
+            input: '<a href="https://www.odoo.com">https://www.odoo.com</a>',
+            output: `&lt;a href="https://www.<span class="${HIGHLIGHT_CLASS}">odoo</span>.com"&gt;https://www.<span class="${HIGHLIGHT_CLASS}">odoo</span>.com&lt;/a&gt;`,
+            searchTerm: "odoo",
```

Comment:
Note: extra coverage for the fix: non-markup are escaped, and search applies to whole content (ignoring tags)